### PR TITLE
Cls2 541 viewing my tasks 2

### PR DESCRIPTION
**Pull request by PaulWheatcroft**
## Description of change

Build a My Tasks table within the My tasks tab on the dashboard to display all tasks associated with an adviser. This list should display the due date in ascending order

## Test instructions

Run this branch pointing to Dev. The dashboard should display the My tasks tab and within that a table of all tasks associated to the logged in user. This will be both assigned to tasks and task that have been created by.

## Screenshots

Screenshot 2023-11-29 at 16 56 26

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to feature branch?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: src/client/components/Dashboard/my-tasks/tasks.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { apiProxyAxios } from '../../Task/utils'
+
+export const getMyTasks = ({ adviser }) =>
+  apiProxyAxios
+    .post('/v4/search/task', {
```
Comment (chopkinsmade) [inline src/client/components/Dashboard/my-tasks/tasks.js:5]: Limiting to 10, AC says 50

File: src/client/components/Dashboard/my-tasks/tasks.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+      limit: 10,
+      offset: 0,
+      created_by: adviser.id,
+      adviser: [adviser.id],
```
Comment (chopkinsmade) [inline src/client/components/Dashboard/my-tasks/tasks.js:9]: Not providing any ordering critera

File: src/client/components/Dashboard/my-tasks/MyTasks.jsx [hunk 3]
```diff
@@ -1,8 +1,27 @@
...
 import { connect } from 'react-redux'
 
-import { state2props } from './state'
+import { GET_MY_TASKS_ID, TASK_GET_MY_TASKS, state2props } from './state'
+import { MY_TASKS_LOADED } from '../../../actions'
+import Task from '../../Task'
+import { MyTasksTable } from './MyTasksTable'
 
-const MyTasks = () => <h1>Hello Tasks</h1>
+const MyTasks = ({ adviser, myTasks }) => {
```
Comment (chopkinsmade) [inline src/client/components/Dashboard/my-tasks/MyTasks.jsx:9]: Can change this to an implicit return, and the <MyTasksTable component can be moved as a child of Task to get rid of the redundant `<>`. Search the codebase for how to render children of tasks, as it's slightly different

File: src/client/components/Dashboard/my-tasks/tasks.js (hunk 1 above)
Comment (PaulWheatcroft) [reply src/client/components/Dashboard/my-tasks/tasks.js:5]: Done

File: src/client/components/Dashboard/my-tasks/tasks.js (hunk 2 above)
Comment (PaulWheatcroft) [reply src/client/components/Dashboard/my-tasks/tasks.js:9]: Ordering now added

Comment (PaulWheatcroft) [reply src/client/components/Dashboard/my-tasks/tasks.js:9]: Ordering now added

File: src/client/components/Dashboard/my-tasks/MyTasks.jsx (hunk 3 above)
Comment (PaulWheatcroft) [reply src/client/components/Dashboard/my-tasks/MyTasks.jsx]: Refactored

File: src/client/components/Dashboard/my-tasks/MyTasksTable.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        </Link>
+      </Table.Cell>
+      <Table.Cell setWidth="20%">{task.investment_project.name}</Table.Cell>
+      <Table.Cell setWidth="25%">{task.advisers[0].name}</Table.Cell>
+    </Table.Row>
+  ))
+}
+
+const MyTasksTable = ({ myTasks }) => (
+  <Table head={header} myTasks={myTasks} data-test="my-tasks-table">
```
[the remark this answers is not in the document]
Comment (PaulWheatcroft) [reply src/client/components/Dashboard/my-tasks/MyTasksTable.jsx:42]: It does not

File: test/component/cypress/specs/Dashboard/MyTasks/MyTasksTable.cy.jsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import { formatMediumDate } from '../../../../../../src/client/utils/date'
+import { MyTasksContent } from '../../../../../../src/client/components/Dashboard/my-tasks/MyTasks'
+import urls from '../../../../../../src/lib/urls'
+
+import { keysToSnakeCase } from '../../../../../functional/cypress/fakers/utils'
+
+describe('My Tasks on the Dashboard', () => {
+  const Component = (props) => <MyTasksContent {...props} />
+  const myTasksList = taskWithInvestmentProjectListFaker()
+  let myTaskResults = []
```
[the remark this answers is not in the document]
Comment (PaulWheatcroft) [reply test/component/cypress/specs/Dashboard/MyTasks/MyTasksTable.cy.jsx:18]: Refactored

File: src/client/components/Dashboard/my-tasks/MyTasksTable.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+const header = (
+  <Table.Row>
+    <StyledTableCellHeader>Date due</StyledTableCellHeader>
+    <StyledTableCellHeader>Task title</StyledTableCellHeader>
+    <StyledTableCellHeader>Project</StyledTableCellHeader>
+    <StyledTableCellHeader>Assigned to</StyledTableCellHeader>
+  </Table.Row>
+)
+
+const rows = (myTasks) => {
```
[the remark this answers is not in the document]
Comment (PaulWheatcroft) [reply src/client/components/Dashboard/my-tasks/MyTasksTable.jsx:23]: Done

**Review by chopkinsmade — APPROVED** [review]

**Pull request merged**